### PR TITLE
Adding support for LTC2688

### PR DESCRIPTION
**Pull request by jsanbuen**
# Description

Created and tested Python Class, Example, and Test Scripts for LTC2688.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Hardware used is the EVAL-CN0554-RPIZ using the LTC2688 specific overlay, effectively disabling the onboard AD7124 and leaving LTC2688 as the only detectable IIO device. Testing and development of python codes were done on a desktop running Ubuntu OS. LAN connector on RPI connected directly to a LAN port on the same desktop unit.

**Test Configuration**:
* Hardware: RPI + EVAL-CN0554-RPIZ (Device Tree Overlay used is for LTC2688)
* OS: Ubuntu

# Documentation

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tfcollins** [comment]
You have 149 files in the change list. Please only update the new files you have created and files you have actually updated.

**Review by tfcollins — CHANGES_REQUESTED** [review]
With all these extra formatting changes my guess is that you are using the wrong version of black. Please use pre-commit

**Comment by jsanbuen** [comment]
Added emu files to the necessary folders. Tried rerunning the pip install for the requirements. Running the "pre-commit run --all-files" keeps on resulting to the lots of additional modified files. The requirements_devs.txt for my forked repo and master is similar (black, no version specified).

**Comment by tfcollins** [comment]
Black is only defined in the pre-commit config: https://github.com/analogdevicesinc/pyadi-iio/blob/master/.pre-commit-config.yaml#L29

You can see CI is changing things back: https://github.com/analogdevicesinc/pyadi-iio/actions/runs/4300046681/jobs/7510479252#step:5:60

**Comment by tfcollins** [comment]
Doc the doc error, rebase on master to get the fix. That was an upstream issue.

For the testing, since the device does not support buffers you probably want to use attribute testing instead like: https://github.com/analogdevicesinc/pyadi-iio/commit/004936a2ba729f8b4f56e0c17c2ab0f611433d02#diff-70dd15dfb722ab21a9724baa86d879d0aa9614fa42e7e04d2d0cdf61065962df

**Comment by tfcollins** [comment]
Alright. We are in good shape now. The only thing left is to fill in the remaining docstrings

**Comment by jsanbuen** [comment]
> Alright. We are in good shape now. The only thing left is to fill in the remaining docstrings

Hi Travis, 

What would I need to add the remaining docstrings? Is there a reference I can check regarding this one?

**Comment by tfcollins** [comment]
These are just comments in each property to say what it does. Most of your dynamic channels are not documented.

**Comment by jsanbuen** [comment]
Hi Travis,

I have already added docstring comments on the remaining channel properties for the LTC2688 class.

**Comment by jsanbuen** [comment]
Hi Travis,

Where can I see your question?

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/ltc2688.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+            """ LTC2688 channel toggle state 1 voltage value setter """
+            self.raw1 = int(((val / self.scale) - self.offset) * (ltc2688.vref / 4.096))
+
+    class _channel_sw_toggle(_channel_toggle):
+        def __init__(self, ctrl, channel_name):
+            super().__init__(ctrl, channel_name)
+
+        @property
+        def symbol(self):
+            """ LTC2688 SW toggle enable flag """
```
[the remark this answers is not in the document]
Comment (jsanbuen) [reply adi/ltc2688.py:265]: The symbol attribute is based on the LTC2688 Linux driver (https://github.com/analogdevicesinc/linux/blob/master/drivers/iio/dac/ltc2688.c). The word "symbol" is technically not found on the LTC2688 datasheet and was just based on the attribute name set on the mentioned driver. In the datasheet, this corresponds to the software toggle.

[the remark this answers is not in the document]
Comment (tfcollins) [reply adi/ltc2688.py:265]: @nunojsa is the word symbol typical here?

**Comment by catapangan** [comment]
Changed name of symbol property in class and example code to toggle_state.

**Comment by jsanbuen** [comment]
Changed "symbol" to "toggle_state" in ltc2688 class and example python scripts.

**Pull request merged**